packages/ra-ui-materialui/src/input/SelectInput.tsx

```diff
@@ -161,6 +162,16 @@ export const SelectInput = (props: SelectInputProps) => {
         source: sourceProp,
     });
 
+    warning(
+        source === undefined,
```

Comment:
This is wrong, as `choices` will always be undefined here

Reply:
Both `source` and `allChoices` variables are set by `useChoicesContext` just above this line. `allChoices` is set asynchrounously (return value of useList hook) so there we have to check for `isLoading !== true` before we throw. However, the `source` variable is set synchronously either by prop or from `ChoicesContext` and should be defined here.